common/lib/capa/capa/capa_problem.py

```diff
@@ -252,21 +252,22 @@ def make_xml_compatible(self, tree):
             if not answer and text:  # trigger of old->new conversion
                 additional.set('answer', text)
                 additional.text = ''
-
         for optioninput in tree.xpath('//optioninput'):
-            correct_option = None
+            correct_option = []
             child_options = []
             for option_element in optioninput.findall('./option'):
```

Comment:
This can be simplified. 
```
correct_elements = [
    option_element.get('correct').upper() == 'TRUE' 
    for option_element in optioninput.findall('./option')
]
if correct_elements.count(True) > 1:
    raise responsetypes.LoncapaProblemError("Multiple Correct values not allowed")
```

Reply:
Not likely. Another list, `child_options` is being maintained at the same time, which contains all the options' names in a specific format. Converting it into list comprehension will require some more changes than you suggested. Moreover, it will affect the code readability.